web/pdf_page_view.js

```diff
@@ -284,6 +288,34 @@ class PDFPageView {
     }
   }
 
+  async #renderTextLayer() {
+    const { pdfPage, textLayer, viewport } = this;
+    if (!textLayer) {
+      return;
+    }
+
+    let error = null;
+    try {
+      if (!textLayer.renderingDone) {
+        const readableStream = pdfPage.streamTextContent({
+          includeMarkedContent: true,
+        });
+        textLayer.setTextContentStream(readableStream);
+      }
+      await textLayer.render(viewport);
+    } catch (ex) {
+      console.error(`#renderTextLayer: "${ex}".`);
+      error = ex;
+    } finally {
+      this.eventBus.dispatch("textlayerrendered", {
+        source: this,
+        pageNumber: this.id,
+        numTextDivs: textLayer.numTextDivs,
+        error,
+      });
+    }
```

Comment:
This would print an error message *and* dispatch the event when textLayer rendering is cancelled, which we don't want here.
To fix that, the easiest solution would probably be to change https://github.com/mozilla/pdf.js/blob/8fda3f04fe290a6ed7af0f684cdef7adea8cfcbe/src/display/text_layer.js#L300 into
```javascript
this._capability.reject(new AbortException("TextLayer task cancelled.")); 
```
and then change the code in this method as follows (you'd probably need to export `AbortException` in [src/pdf.js](https://github.com/mozilla/pdf.js/blob/master/src/pdf.js) as well).
```suggestion
    } catch (ex) {
      if (ex instance AbortException) {
        return;
      }
      console.error(`#renderTextLayer: "${ex}".`);
      error = ex;
    }
    this.eventBus.dispatch("textlayerrendered", {
      source: this,
      pageNumber: this.id,
      numTextDivs: textLayer.numTextDivs,
      error,
    });
```